src/memory/sqlite_store.rs

```diff
@@ -0,0 +1,225 @@
+use crate::{Address, MemoryADT};
+use r2d2::Pool;
+use r2d2_sqlite::SqliteConnectionManager;
+use rusqlite::{params_from_iter, types::ToSqlOutput};
+use std::{marker::PhantomData, thread::sleep, time::Duration};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum SqlMemoryError {
+    #[error("Sql store memory error: {0}")]
+    SqlError(#[from] rusqlite::Error),
+    #[error("r2d2 pooling error: {0}")]
+    PoolingError(#[from] r2d2::Error),
+}
+
+#[derive(Debug, Clone)]
+pub struct SqlMemory<Address, Word> {
+    pool: Pool<SqliteConnectionManager>,
+    // only enable this in case of a high contention scenario
+    exponential_backoff: bool,
+    _marker: PhantomData<(Address, Word)>,
+}
+
+const MAX_POOL_SIZE: u32 = 10; // placeholder
+const CREATE_TABLE_SCRIPT: &str = "
+CREATE TABLE IF NOT EXISTS findex_aw_store
+(
+    adr BLOB PRIMARY KEY,
```

Comment:
please no spelling upon abbreviating: either use `(a, w)` as usual or `(addr, word)` if you really want to innovate.